gbpservice/nfp/base_configurator/api/v1/controllers/controller.py

```diff
@@ -69,19 +86,36 @@ def get(self):
         Returns: Dictionary that contains Notification data
 
         """
-
+        global cache_ips
         global notifications
         try:
-            notification_data = jsonutils.dumps(notifications)
-            msg = ("NOTIFICATION_DATA sent to config_agent %s"
-                   % notification_data)
-            LOG.info(msg)
-            notifications = []
-            return notification_data
+            if not cache_ips:
+                notification_data = jsonutils.dumps(notifications)
+                msg = ("NOTIFICATION_DATA sent to config_agent %s"
+                       % notification_data)
+                LOG.info(msg)
+                notifications = []
+                return notification_data
+            else:
+                for ip in cache_ips:
+                    notification_response = requests.get(
+                        'http://'+str(ip)+':8080/v1/nfp/get_notifications')
+                    notification = jsonutils.loads(notification_response.text)
+                    notifications.extend(notification)
+                    cache_ips.remove(ip)
```

Comment:
If two POST requests come at a time, we are adding the device_ip only once for these. But, while pulling notifications, we are deleting the device_ip from the cache. This will result in a case where the second POST response will not be pulled from over the cloud service VM.